libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1364,8 +1362,7 @@ void SMTEncoder::indexOrMemberAssignment(Expression const& _expr, smtutil::Expre
 			if (varDecl->hasReferenceOrMappingType())
 				resetReferences(*varDecl);
 
-			m_context.addAssertion(m_context.newValue(*varDecl) == toStore);
-			m_context.expression(*id)->increaseIndex();
```

Comment:
I think we should keep this line, as it's not done by the new call

Reply:
The same reasoning as in the comment above. The increment of the index is done in the `defineExpr` on the next line, if necessary.